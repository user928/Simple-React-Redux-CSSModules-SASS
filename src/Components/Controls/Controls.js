import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { decrementNum, incrementNum, resetCounter } from '../../Redux/actionCreators/counterActions';
import styles from './Controls.module.css';

export class Controls extends React.Component {
	static propTypes = {
		increment: PropTypes.func,
		decrement: PropTypes.func,
		resetCount: PropTypes.func,
		resetLabel: PropTypes.string,
		incrementLabel: PropTypes.string,
		decrementLabel: PropTypes.string,
	};

	render() {
		return (
			<div className={styles.controls}>
				<div className={styles.controlAction} data-ui="decrementBtn" onClick={this.props.decrement}>
					-
				</div>

				<div className={styles.controlAction} data-ui="resetBtn" onClick={this.props.resetCount}>
					Reset
				</div>

				<div className={styles.controlAction} data-ui="incrementBtn" onClick={this.props.increment}>
					+
				</div>
			</div>
		);
	}
}

export const mapDispatchToProps = dispatch => ({
	increment: () => dispatch(incrementNum()),
	decrement: () => dispatch(decrementNum()),
	resetCount: () => dispatch(resetCounter()),
});

export default connect(
	null,
	mapDispatchToProps
)(Controls);
