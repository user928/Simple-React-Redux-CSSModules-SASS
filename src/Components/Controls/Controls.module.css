/* Controls styles */
@value variables: "../../styles/shared-styles/shared-styles.css";
@value --red from variables;

.controls {
	display: flex;
}

.controlAction {
	composes: paddingALL--xxs noMargin from '../../styles/shared-styles/shared-styles.css';
	min-width: 40px;
	text-align: center;
	background: #fff;
	color: --red;
	display: flex;
	align-items: center;
	justify-content: center;
	margin: 0 0 0 5px;
	cursor: pointer;
}
