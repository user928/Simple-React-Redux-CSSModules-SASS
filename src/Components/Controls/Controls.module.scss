/* Controls styles */
@import '../../styles/shared-styles/variables'; //TODO find a way not to have relative imports...

.controls {
	display: flex;
}

.controlAction {
	composes: paddingALL--sm noMargin from '../../styles/shared-styles/shared-styles.css'; //TODO find a way not to have relative imports...
	min-width: 40px;
	text-align: center;
	background: #fff;
	color: $red;
	display: flex;
	align-items: center;
	justify-content: center;
	margin: 0 0 0 5px;
	cursor: pointer;
}
