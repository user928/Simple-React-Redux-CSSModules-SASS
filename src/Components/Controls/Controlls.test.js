import { Controls } from './Controls';

describe('Controls component', () => {
	it('renders correctly', () => {
		const wrapper = render(<Controls />);

		expect(wrapper).toMatchSnapshot();
	});

	it('find reset txt', () => {
		const wrapper = mount(<Controls />);
		const resetTxt = wrapper.find('[data-ui="resetBtn"]').text();
		expect(resetTxt).toBe('Reset');
	});

	it('click on increment once', () => {
		const spy = sinon.spy();
		const wrapper = mount(<Controls increment={spy} />);

		wrapper.find('[data-ui="incrementBtn"]').simulate('click');
		expect(spy.calledOnce).toBe(true);
	});
});
