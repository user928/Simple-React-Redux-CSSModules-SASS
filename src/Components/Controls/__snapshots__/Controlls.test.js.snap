// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders correctly 1`] = `
<div
  class="controls"
>
  <div
    class="controlAction"
    data-ui="decrementBtn"
  >
    -
  </div>
  <div
    class="controlAction"
    data-ui="resetBtn"
  >
    Reset
  </div>
  <div
    class="controlAction"
    data-ui="incrementBtn"
  >
    +
  </div>
</div>
`;
